typing/env.ml

```diff
@@ -2846,6 +2853,36 @@ let add_exclave_lock env = add_lock Exclave_lock env
 
 let add_unboxed_lock env = add_lock Unboxed_lock env
 
+let add_local_env_lock env = add_lock Local_env_lock env
+
+let add_quotation_lock env =
+  add_lock Quotation_lock {env with stage = env.stage + 1}
```

Comment:
Better call this `enter_quatation`, since it does more than just adding a quation lock.
(similarly, below should be called `enter_splice`).